packages/shopify-codemod/transforms/splitReturnAssignments.test.js

```diff
@@ -0,0 +1,37 @@
+import {matchLast} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  return j(source)
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: {
+            type: 'AssignmentExpression',
+          },
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+
+      const {left, right} = body[body.length - 1].argument;
+      delete body[body.length - 1];
```

Comment:
Assigning the return value of `body.pop` to `{left, right}` is very clever! Updated.